### PR TITLE
Enable several variants for HDF5

### PR DESCRIPTION
**Pull request by eschnett**
+cxx, +fortran: Enable support for the respective languages
+threadsafe: Enable multi-threading

### CODE REVIEW

File: var/spack/packages/hdf5/package.py [hunk 1]
```diff
@@ -10,12 +10,15 @@ class Hdf5(Package):
...
     list_url = "http://www.hdfgroup.org/ftp/HDF5/releases"
     list_depth = 3
-    
+
     version('1.8.16', 'b8ed9a36ae142317f88b0c7ef4b9c618')
     version('1.8.15', '03cccb5b33dbe975fdcd8ae9dc021f24')
     version('1.8.13', 'c03426e9e77d7766944654280b467289')
 
+    variant('cxx', default=False, description='Enable C++ support')
+    variant('fortran', default=False, description='Enable Fortran support')
```
Comment (tgamblin) [inline var/spack/packages/hdf5/package.py]: Can you make these two enabled by default?  I think in the common case we might as well build the C++ and Fortran interfaces, unless there's a good reason not to.  @markcmiller86 -- opinions?

**Comment by eschnett** [comment]
I think this is a good idea. Done.

File: var/spack/packages/hdf5/package.py (hunk 1 above)
Comment (markcmiller86) [inline var/spack/packages/hdf5/package.py]: FYI…

I think c++ and fortran interfaces to HDF5 are less well used/tested than C interface. But, thats just my opinion and not based on any real usage information.

Nonetheless, I opted to default these to off to minimize likelihood of failure to build HDF5 as other packages I am aware of that require it don't need 'em.

But, these opinions are just based on being cautious and not ever really having needed c++ or fortran interfaces to HDF5.

Mark

**Comment by eschnett** [comment]
Ping

**Pull request merged**